service/history/shard/context_impl.go

```diff
@@ -1889,10 +1890,13 @@ func (s *ContextImpl) acquireShard() {
 	// lifecycleCtx. The persistence operations called here use lifecycleCtx as their context,
 	// so if we were blocked in any of them, they should return immediately with a context
 	// canceled error.
-	policy := backoff.NewExponentialRetryPolicy(1 * time.Second).
-		WithExpirationInterval(5 * time.Minute)
-
 	// Remember this value across attempts
+	policy := s.acquireShardRetryPolicy
+	if policy == nil {
+		policy = backoff.NewExponentialRetryPolicy(time.Nanosecond).
```

Comment:
Nanosecond could be too short.

Reply:
Default policy should be the one with 5m retry. Only overwrite acquireShardRetryPolicy in test. The acquireShardRetryPolicy could even be named as acquireShardRetryPolicyForTest. Or move this retry policy to dynamic config and put 5m retry as default then overwrite dynamic config in test.

Reply:
Fixed